packages/core/src/browser/decoration-style.ts

```diff
@@ -0,0 +1,72 @@
+/********************************************************************************
+ * Copyright (C) 2021 Ericsson and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+export namespace DecorationStyle {
+
+    export function createStyleSheet(container: HTMLElement = document.getElementsByTagName('head')[0]): CSSStyleSheet | undefined {
+        if (!container) {
+            return undefined;
+        }
+        const style = document.createElement('style');
+        style.id = 'editorDecorationsStyle';
+        style.type = 'text/css';
+        style.media = 'screen';
+        style.appendChild(document.createTextNode('')); // trick for webkit
+        container.appendChild(style);
+        return <CSSStyleSheet>style.sheet;
+    }
+
+    function getRuleIndex(selector: string, styleSheet: CSSStyleSheet): number | undefined {
+        const rules = styleSheet.cssRules || styleSheet.rules;
+        for (let i = 0; i < rules.length; i++) {
+            if (rules[i].type === CSSRule.STYLE_RULE && (<CSSStyleRule>rules[i]).selectorText === selector) {
+                return i;
+            }
+        }
+    }
+
+    export function createStyleRule(selector: string, styleSheet: CSSStyleSheet | undefined): CSSStyleRule | void {
+        if (!styleSheet) {
+            return;
+        }
+
+        let index = getRuleIndex(selector, styleSheet);
+        if (!index) {
```

Comment:
If  you adopt the `Array.from` suggestion above, this should be `index !== -1`. If not, this should be `index !== undefined` since `!0` would return `true`, but `0` is a real index, so we wouldn't need to create a new rule.